FG addon nasal radios rewrite

colingeniet opened the pull request:
Fixes #117 : make sure COM radios unused by the aircraft only transmit empty values to the plugin.
This also avoids modifying properties which are not in a `fgcom-mumble` property subtree (e.g. for ADF, `frequencies/selected-khz` is no longer cleared if the ADF is unused. Instead it is `fgcom-mumble/selected-mhz` which remains empty).

Some remarks on the code side:
- There is a new `GenericRadio` class. Almost the only thing it does it decide whether or not a radio is used (still by testing if `operable` exists). It sets an attribute `is_used` to indicate this.
- For ADFs, the listeners/timers are not set up if the radio is not used. Thus `recalc{Volume,Frequency}` are written assuming that the radio is used, and will always set a non-empty value to their output.

Last thing I want to do is remove unused radios from the COM bar.

hbeni commented:
Great to see your progress!

> Last thing I want to do is remove unused radios from the COM bar.

Have you seen my comment with the idea here: https://github.com/hbeni/fgcom-mumble/issues/117#issuecomment-843491825
Of course, now that you have the `is-used` property, this might be way easier to use.

hbeni commented on client/fgfs-addon/addon-main.nas:
```diff
@@ -7,6 +7,56 @@
+        # in the fgcom-mumble plugin), test if the 'operable' property exist.
+        # It is created by the C++ instrument code, if the radio is used.
+        me.operable = me.root.getNode("operable");
+        me.is_used = (me.operable != nil);
```
Consider publishing this as property in the fgcom-mumble subnode, so the combar can pick it up easily :)

colingeniet commented:
This is ready for merge for me.

Additional changes:
- Moved the radio objects + RDF code to `radio.nas`. This way `addon-main.nas` only does plugin initialization / shutdown
- Added destructors  and clean-up functions in `radio.nas`. They are not used yet, but I think it's better to have them.
- Added the `is-used` property as suggested, used for the COM bar.

hbeni commented:
Looks very good code wise :)

Let me do some additional testing.

hbeni commented:
Awesome work! Thank you very much! :partying_face:

hbeni commented:
For the archives: The dragon did not work, it registered two COMs and one ADF.
I believe this is a "problem" in the C++ code of FGFS however - it really looks like, if no radio is defined in the aircraft, it just defaults to initializing COM1+COM2+ADF1. So nothing we should worry about.

colingeniet commented:
> I believe this is a "problem" in the C++ code of FGFS however - it really looks like, if no radio is defined in the aircraft, it just defaults to initializing COM1+COM2+ADF1.

It's not C++ actually: `FGData/Aircraft/Generic/generic-instrumentation.xml` is loaded by default in `FGData/default.xml`.
